clang/include/clang/AST/Type.h

```diff
@@ -4914,6 +4920,78 @@ class FunctionEffectsRef {
   void dump(llvm::raw_ostream &OS) const;
 };
 
+/// A mutable set of FunctionEffect::Kind.
+class FunctionEffectKindSet {
```

Comment:
I agree that this can and should use `std::bitset` instead of reinventing it, as a representation, but that doesn't help with the problem of iterating through the set bits, and a lot of code wants to iterate through these sets.

The only alternative I can think of is a `for_each()` method. Sometimes that can be messier than it's worth and sometimes it is better than a special iterator. Will try it.

Reply:
> but that doesn't help with the problem of iterating through the set bits

Ah, I missed the fact that `std::bitset` doesn’t have iterators. I also just saw that the iteration logic is also a bit more advanced than just returning every element, so that’ll probably have to stay. 

(I can probably think of a way to mess around w/ `filter`, `enumerate` and `map_range` to get this to work if we had a bitset iterator, but I’m not sure it’s worth the effort; there is `llvm::Bitset`, but that doesn’t have iterators either...)

Reply:
For now I've left the special iterator but replaced the manual bit-twiddling with a `std::bitset`.